src/main/java/org/apache/fineract/utils/CsvUtility.java

```diff
@@ -0,0 +1,43 @@
+package org.apache.fineract.utils;
+
+import org.apache.fineract.operations.TransactionRequest;
+import org.supercsv.io.CsvBeanWriter;
+import org.supercsv.io.ICsvBeanWriter;
+import org.supercsv.prefs.CsvPreference;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+public class CsvUtility {
+
+    public static void writeToCsv(HttpServletResponse response, List<TransactionRequest> transactionRequests) throws IOException {
```

Comment:
This function is very large. Please break it down into multiple functions.

Reply:
noted

Reply:
The new implementation has the below design
1. A new generic class [CsvWriter] for writing a list of POJOs into CSV.
2. Below is the use case
```
CsvWriter<TransactionRequest> writer = new CsvWriter.Builder<TransactionRequest>()
                .setHeader(getCsvHeader(transactionRequestsFields))
                .setNameMapping(transactionRequestsFields)
                .setPrintWriter(response)
                .setData(transactionRequests)
                .build();

writer.write();
```